app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -461,44 +470,38 @@ class YearToDateLiabilityController @Inject() (
   def personalAllowanceSubmit(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
     withFillingOutReturnAndYTDLiabilityAnswers(request) { (_, fillingOutReturn, answers) =>
       (answers, fillingOutReturn.draftReturn) match {
+        case (_, _) if isATrust(fillingOutReturn) =>
+          Redirect(routes.YearToDateLiabilityController.checkYourAnswers())
         case (calculatedAnswers: CalculatedYTDAnswers, draftReturn: DraftSingleDisposalReturn) =>
           withDisposalDate(draftReturn) { disposalDate =>
             withEstimatedIncome(calculatedAnswers) { estimatedIncome =>
               if (estimatedIncome.value > 0L) {
                 commonSubmitBehaviour(fillingOutReturn, draftReturn, calculatedAnswers)(
                   form = personalAllowanceForm(disposalDate)
-                )(page = { (form, backLink) =>
-                  val updatedForm = form.copy(errors = form.errors.map(
-                    _.copy(args = Seq(
-                      MoneyUtils
-                        .formatAmountOfMoneyWithoutPoundSign(
-                          disposalDate.taxYear.personalAllowance.inPounds()
-                        )
-                    )
-                    )
-                  )
-                  )
-
-                  personalAllowancePage(updatedForm, backLink, disposalDate)
+                )(page = {
+                  case (form, backLink) =>
+                    personalAllowancePage(form, backLink, disposalDate)
                 })(
                   requiredPreviousAnswer = _.fold(_.estimatedIncome, c => Some(c.estimatedIncome)),
                   routes.YearToDateLiabilityController.estimatedIncome()
-                ) { (p, draftReturn) =>
-                  val personalAllowance = AmountInPence.fromPounds(p)
-                  if (calculatedAnswers
-                        .fold(_.personalAllowance, _.personalAllowance)
-                        .contains(personalAllowance)) {
-                    draftReturn
-                  } else {
-                    val newAnswers = calculatedAnswers
-                      .unset(_.hasEstimatedDetails)
-                      .unset(_.calculatedTaxDue)
-                      .unset(_.taxDue)
-                      .unset(_.mandatoryEvidence)
-                      .copy(personalAllowance = Some(AmountInPence.fromPounds(p)))
+                ) {
+                  case (p, draftReturn) =>
```

Comment:
there are these `cases` as well which you don't need

Reply:
All done, thanks